lib/spack/spack/ci.py

```diff
@@ -2020,50 +2035,66 @@ def process_command(name, commands, repro_dir, run=True, exit_on_failure=True):
 
     Returns: the exit code from processing the command
     """
-    tty.debug("spack {0} arguments: {1}".format(name, commands))
 
+    tty.debug(f"spack {name} arguments: {commands}")
     if len(commands) == 0 or isinstance(commands[0], str):
         commands = [commands]
-
-    # Create a string [command 1] && [command 2] && ... && [command n] with commands
-    # quoted using double quotes.
-    args_to_string = lambda args: " ".join('"{}"'.format(arg) for arg in args)
-    full_command = " \n ".join(map(args_to_string, commands))
-
-    # Write the command to a shell script
-    script = "{0}.sh".format(name)
+    ext = "ps1" if IS_WINDOWS else "sh"
+    verbose_commands = "Set-PSDebug -Trace 2\n" if IS_WINDOWS else "set -x\n"
+    shebang = "" if IS_WINDOWS else "#!/bin/sh\n\n"
+    exit_error = '$ErrorActionPreference = "Stop"\n' if IS_WINDOWS else "set -e\n"
+    # ErrorActionPreference will handle PWSH commandlets (Spack calls),
+    # but we need to handle EXEs (git, etc) ourselves
+    catch_exe_failure = (
+        """
+if ($LASTEXITCODE -ne 0){
+    throw "Command {} has failed"
+}
+"""
+        if IS_WINDOWS
+        else ""
+    )
+    # Create a string [command 1] \n [command 2] \n ... \n [command n] with
+    # commands composed into a platform dependent shell script, pwsh on Windows,
+    args_to_string = lambda args: f'{" ".join(args)}\n{catch_exe_failure.format(" ".join(args))}'
+    full_command = "\n".join(map(args_to_string, commands))
+    # Write the command to a python script
+    script = f"{name}.{ext}"
     with open(script, "w") as fd:
```

Comment:
Since every line here is abstracted, IMO it would be more clear to split on `IS_WINDOWS` here like:

```
if IS_WINDOWS:
    script = [
        ...
    ]
else:
    script = [
        ...
    ]
```

Reply:
Sounds good, will refactor.